tests/drivers/i2s/i2s_speed: mimxrt1189/cm33: build fails due to CONFIG_DMA_TCD_QUEUE_SIZE too small for i2s_mcux_sai

### Describe the bug

this is found in customized board testing we can name it virtual_board

`tests/drivers/i2s/i2s_speed` fails to build for `virtual_board/mimxrt1189/cm33` because the MCUX SAI I2S driver enforces `CONFIG_DMA_TCD_QUEUE_SIZE > 3`, but the MCUX eDMA Kconfig default is `2`.

The failure occurs at compile time in `drivers/i2s/i2s_mcux_sai.c`:

```c
#define NUM_DMA_BLOCKS_RX_PREP 3
#define MAX_TX_DMA_BLOCKS CONFIG_DMA_TCD_QUEUE_SIZE
BUILD_ASSERT(MAX_TX_DMA_BLOCKS > NUM_DMA_BLOCKS_RX_PREP,
            "NUM_DMA_BLOCKS_RX_PREP must be < CONFIG_DMA_TCD_QUEUE_SIZE");
```

### To reproduce
1. Build the test:
   - Test: `drivers.i2s.speed.drivers_i2s_speed_both_rxtx.i2s_dir_both_transfer_long_44100`
   - Platform: `mimxrt1189/cm33`
2. Observe build failure in `drivers/i2s/i2s_mcux_sai.c`.

### Build log excerpt
```
.../zephyr/include/zephyr/toolchain/gcc.h:87:36: error: static assertion failed: "NUM_DMA_BLOCKS_RX_PREP must be < CONFIG_DMA_TCD_QUEUE_SIZE"
.../zephyr/drivers/i2s/i2s_mcux_sai.c:44:1: note: in expansion of macro 'BUILD_ASSERT'
```

### Environment
- Zephyr: `v4.3.0-5938-gd5f09c1b0464`
- Toolchain: GCC (as used by CI)

### Expected behavior
The `i2s_speed` test should build on `nxp_virtual/mimxrt1189/cm33` without requiring manual Kconfig overrides.

### Proposed fix
Update the test configuration to ensure an adequate TCD queue size when using MCUX eDMA + MCUX SAI I2S, e.g.:

- Add to `tests/drivers/i2s/i2s_speed/prj.conf` (simple but may cause “undefined symbol” warnings on non-MCUX platforms):
  ```conf
  CONFIG_DMA_TCD_QUEUE_SIZE=4
  ```

or (preferred to avoid affecting other platforms)

- Add a board-specific test config file for the affected platform (e.g. `tests/drivers/i2s/i2s_speed/boards/nxp_virtual_mimxrt1189_cm33.conf`) containing:
  ```conf
  CONFIG_DMA_TCD_QUEUE_SIZE=4
  ```

### version:

os: ubuntu
toolchain: zephyr
version: 2fd6d9ad8d36918d98fd2747a9b4fb8f3c07c971 

### Additional context
`drivers/dma/Kconfig.mcux_edma` currently defaults `CONFIG_DMA_TCD_QUEUE_SIZE` to `2`, which is incompatible with the current constraints in `drivers/i2s/i2s_mcux_sai.c` for this test scenario.

--- a/drivers/dma/Kconfig.mcux_edma
+++ b/drivers/dma/Kconfig.mcux_edma
@@ -50,9 +50,13 @@ if DMA_MCUX_EDMA || DMA_MCUX_EDMA_V3 || DMA_MCUX_EDMA_V4
 
 config DMA_TCD_QUEUE_SIZE
 	int "number of TCD in a queue for SG mode"
+	default 4 if I2S_MCUX_SAI
 	default 2
 	help
-	  number of TCD in a queue for SG mode
+	  Number of TCDs in a queue for SG mode.
+	  The MCUX SAI I2S driver preloads 3 RX DMA blocks and also requires
+	  additional queue space for TX, so it needs this queue size to be
+	  greater than 3.
 
 config DMA_MCUX_TEST_SLOT_START
 	int "test slot start num"

